### PR TITLE
chore(Carta Giovani Nazionale): [#179247891] Sync cgn updated specs

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR sync codebase with cgn updated specs

### CODE REVIEW

File: ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx [hunk 1]
```diff
@@ -78,19 +79,22 @@ const CgnDiscountDetail: React.FunctionComponent<Props> = ({
...
-              size={CATEGORY_ICON_SIZE}
-              color={IOColors.bluegrey}
-            />
-            <View hspacer small />
-            <H5 weight={"SemiBold"} color={"bluegrey"}>
-              {I18n.t(c.nameKey).toLocaleUpperCase()}
-            </H5>
-          </View>
-        ))
+        getCategorySpecs(categoryKey as ProductCategoryEnum).fold(
```
Comment (fabriziofff) [inline ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx:82]:
```suggestion
        getCategorySpecs(categoryKey).fold(
```
This should not be required, in my understanding the type seems already inferred. What you think?

File: ts/features/bonus/cgn/components/merchants/CgnMerchantsDiscountItem.tsx [hunk 2]
```diff
@@ -42,19 +43,22 @@ const CgnMerchantDiscountItem: React.FunctionComponent<Props> = ({
...
-                      size={22}
-                      color={IOColors.bluegrey}
-                    />
-                    <View hspacer small />
-                    <H5 weight={"SemiBold"} color={"bluegrey"}>
-                      {I18n.t(c.nameKey).toLocaleUpperCase()}
-                    </H5>
-                  </View>
-                ))
+                getCategorySpecs(categoryKey as ProductCategoryEnum).fold(
```
Comment (fabriziofff) [inline ts/features/bonus/cgn/components/merchants/CgnMerchantsDiscountItem.tsx:46]:
```suggestion
                getCategorySpecs(categoryKey).fold(
```
The same as before, I think this should not be required.

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx (hunk 1 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx]: I handled it cause it was in the output of `tsc:noemit` check...seems not to be required anymore

Comment (CrisTofani) [reply ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx]: ddde264 fixes

File: ts/features/bonus/cgn/components/merchants/CgnMerchantsDiscountItem.tsx (hunk 2 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/components/merchants/CgnMerchantsDiscountItem.tsx]: ddde264 fixes

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**